File: backend/src/main/java/net/pengcook/ingredient/dto/IngredientResponse.java
package net.pengcook.ingredient.dto;

import net.pengcook.ingredient.domain.IngredientRecipe;
import net.pengcook.ingredient.domain.Requirement;

public record IngredientResponse(long ingredientId, String ingredientName, Requirement requirement) {

    public IngredientResponse(IngredientRecipe ingredientRecipe) {
        this(
                ingredientRecipe.getIngredient().getId(),
                ingredientRecipe.getIngredient().getName(),
                ingredientRecipe.getRequirement()
        );
    }
}

File: android/app/src/main/java/net/pengcook/android/data/model/feed/item/IngredientResponse.kt
package net.pengcook.android.data.model.feed.item

import com.google.gson.annotations.SerializedName

data class IngredientResponse(
    @SerializedName("ingredientId")
    val ingredientId: Long,
    @SerializedName("ingredientName")
    val ingredientName: String,
    @SerializedName("requirement")
    val requirement: String,
)
